### PR TITLE
Log name and role claims as information, when running locally

### PR DESCRIPTION
**Pull request by martinnormark**
We're using custom App Roles in our application: https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-add-app-roles-in-azure-ad-apps

Users are granted membership of App Roles in Azure AD, and the membership appears as `roles` claims in the token:

```
         {
            "typ":"roles",
            "val":"Test_Reader"
         },
         {
            "typ":"roles",
            "val":"Test_Admin"
         },
         {
            "typ":"roles",
            "val":"Test_Expert"
         },
```

We had some issues after upgrading this package with new configuration, and it wasn't immediately evident what was missing, as the authentication part itself worked fine, but Authorization via `AuthorizeAttribute` in controllers failed, as we had required roles: `[Authorize(Roles = "Test_Expert")]`

First of all, I've added description of App Roles to the readme to inform about the need to override the claims type of the roles claims, to find the App Roles claims in the token.

Second, I've added more logging to `LocalAuthMeService` to inform about the name and roles claims found in the JSON payload. This would've been helpful to see in the console, to clearly see what goes inside the `AuthenticationTicket` parsed from the JSON payload.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,40 @@ The result of the request is a json with the authentication information of your
 
 If you want to add roles to the `User` property you can have a look in `Transformers/ClaimsTransformer.cs` in the Sample project. There you can see an example how to get started with this.
 
+### Adding App Roles and receiving them in the token
+Azure AD supports [adding custom App Roles](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-add-app-roles-in-azure-ad-apps) to your application manifest, assigning users or groups to the App Roles and [receiving user membership (direct or via groups) in the token as `roles` claims](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-add-app-roles-in-azure-ad-apps#receive-roles-in-tokens).
```
Comment (paule96) [inline README.md:124]: add an empty line between heading and text.

File: src/KK.AspNetCore.EasyAuthAuthentication/Services/LocalAuthMeService.cs [hunk 2]
```diff
@@ -87,14 +87,23 @@ private async Task<AuthenticationTicket> CreateUserTicket()
...
-            return AuthenticationTicketBuilder.Build(
+            var ticket = AuthenticationTicketBuilder.Build(
                     JsonConvert.DeserializeObject<IEnumerable<AADClaimsModel>>(payload["user_claims"].ToString()),
                     providerName,
                     this.defaultOptions.GetProviderOptions()
                 );
+
+            var name = ticket.Principal.Claims?.FirstOrDefault(c => c.Type == ClaimTypes.Name)?.Value ?? string.Empty;
+            var roles = ticket.Principal.Claims?.Where(c => c.Type == ClaimTypes.Role);
+            var rolesString = string.Join(", ", roles.Select(r => r.Value)) ?? string.Empty;
```
Comment (paule96) [inline src/KK.AspNetCore.EasyAuthAuthentication/Services/LocalAuthMeService.cs:102]: `roles` can be `null` here. please add an `?`

**Review by paule96 — CHANGES_REQUESTED** [review]
Looks good. Thanks for your Feedback. 

Only minor adjustments. 🤘

**Comment by martinnormark** [comment]
@paule96 Corrected now, thanks!

**Review by paule96 — APPROVED** [review]

**Comment by kirkone** [comment]
Package [PR-57-KK.AspNetCore.EasyAuthAuthentication](https://dev.azure.com/kirkone/KK.AspNetCore.EasyAuthAuthentication/_packaging?_a=package&feed=KK.AspNetCore.EasyAuthAuthentication-PR&package=PR-57-KK.AspNetCore.EasyAuthAuthentication&protocolType=NuGet) published to [KK.AspNetCore.EasyAuthAuthentication-PR](https://dev.azure.com/kirkone/KK.AspNetCore.EasyAuthAuthentication/_packaging?_a=feed&feed=KK.AspNetCore.EasyAuthAuthentication-PR)

kirkone/KK.AspNetCore.EasyAuthAuthentication

**Pull request merged**